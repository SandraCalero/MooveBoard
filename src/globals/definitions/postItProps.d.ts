export interface IPostIt {
	id: number;
	content?: string;
}

export interface IPostItEditable extends IPostIt {
	disabled?: boolean;
}

export interface IPostItWithEvent extends IPostIt {
	showModal: (postIt: IPostIt) => void;
}

export interface IPostItEditableWithEvent extends IPostItEditable {
	showModal: (postIt: PostIt) => void;
}
