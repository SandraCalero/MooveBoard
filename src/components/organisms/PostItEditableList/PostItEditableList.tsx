import PostIt from 'components/molecules/PostIt/PostIt';
import { IPostItEditable, IPostIt } from 'globals/definitions/postItProps';

interface PostItListProps {
	postItList: Array<IPostItEditable>;
	showModal: (postIt: IPostIt) => void;
}

export default function PostItEditableList({
	postItList,
	showModal,
}: PostItListProps) {
	return (
		<ul
			className='grid grid-cols-auto p-4 m-0 list-none gap-4
			justify-center'
		>
			{postItList.map((postIt) => (
				<li className='justify-self-center' key={postIt.id}>
					<PostIt
						id={postIt.id}
						showModal={showModal}
						content={postIt.content}
						disabled={postIt.disabled}
					/>
				</li>
			))}
		</ul>
	);
}
