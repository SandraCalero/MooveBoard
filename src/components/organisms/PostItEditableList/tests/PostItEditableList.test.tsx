import { IPostItEditable } from 'globals/definitions/postItProps';
import { render } from 'utils/tests/test-utils';
import PostItEditableList from '../PostItEditableList';

describe('<PostItEditableList />', () => {
	it('should match the snapshot', () => {
		const postItList: Array<IPostItEditable> = [
			{
				id: 1,
				content: 'Testing list of post its, this is post it one',
				disabled: true,
			},
			{
				id: 2,
				content: 'Testing list of post its, this is post it two',
				disabled: true,
			},
		];
		const showModal = jest.fn();
		const { container, getByText } = render(
			<PostItEditableList postItList={postItList} showModal={showModal} />
		);
		expect(container).toMatchSnapshot();
		expect(
			getByText('Testing list of post its, this is post it one')
		).toBeTruthy();
		expect(
			getByText('Testing list of post its, this is post it two')
		).toBeTruthy();
	});
});
