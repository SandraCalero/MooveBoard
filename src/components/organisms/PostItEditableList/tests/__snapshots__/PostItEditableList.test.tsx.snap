// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostItEditableList /> should match the snapshot 1`] = `
<div>
  <ul
    class="grid grid-cols-auto p-4 m-0 list-none gap-4 justify-center"
  >
    <li
      class="justify-self-center"
    >
      <div
        aria-hidden="true"
        class="flex flex-col items-center justify-center bg-yellow shadow-xl w-max md:absolute md:cursor-grab dark:bg-cerulean-500 dark:shadow-xl dark:shadow-cerulean-700 transition-colors dark:border dark:border-grey dark:text-mine-shaft-600"
        draggable="true"
        id="1"
      >
        <button
          class="flex flex-col items-center justify-center border-none
			m-2 p-0	bg-transparent hover:opacity-90 active:translate-y-1 my-0 mx-0 px-2 pt-2 opacity-20 text-xl hover:opacity-100 self-end dark:text-mine-shaft-900 dark:opacity-40 dark:hover:opacity-100 transition-colors"
          name="button"
          type="button"
        >
          X
        </button>
        <textarea
          class="p-2 w-52 h-52 bg-transparent outline-none border-none resize-y md:resize md:cursor-grab focus:cursor-text"
          disabled=""
          name="note"
        >
          Testing list of post its, this is post it one
        </textarea>
      </div>
    </li>
    <li
      class="justify-self-center"
    >
      <div
        aria-hidden="true"
        class="flex flex-col items-center justify-center bg-yellow shadow-xl w-max md:absolute md:cursor-grab dark:bg-cerulean-500 dark:shadow-xl dark:shadow-cerulean-700 transition-colors dark:border dark:border-grey dark:text-mine-shaft-600"
        draggable="true"
        id="2"
      >
        <button
          class="flex flex-col items-center justify-center border-none
			m-2 p-0	bg-transparent hover:opacity-90 active:translate-y-1 my-0 mx-0 px-2 pt-2 opacity-20 text-xl hover:opacity-100 self-end dark:text-mine-shaft-900 dark:opacity-40 dark:hover:opacity-100 transition-colors"
          name="button"
          type="button"
        >
          X
        </button>
        <textarea
          class="p-2 w-52 h-52 bg-transparent outline-none border-none resize-y md:resize md:cursor-grab focus:cursor-text"
          disabled=""
          name="note"
        >
          Testing list of post its, this is post it two
        </textarea>
      </div>
    </li>
  </ul>
</div>
`;
